addons/quest_system/quest_manager.gd

```diff
@@ -147,33 +146,61 @@ func call_quest_method(quest_id: int, method: String, args: Array) -> void:
 
 
 func set_quest_property(quest_id: int, property: String, value: Variant) -> void:
-	var quest: Quest = null
-
+	
 	# Find the quest
-	for pools in get_all_pools():
-		if pools.get_quest_from_id(quest_id) != null:
-			quest = pools.get_quest_from_id(quest_id)
+	var quest: Quest = __get_quest_by_id(quest_id)
 
 	if quest == null: return
 
 	# Now check if the quest has the property
+	var property_found : bool = __quest_has_property(quest, property)
+
+	# Return if the property was not found
+	if not property_found: return
 
-	# First if the property is null -> we return
-	if property == null: return
+	# Finally we set the value
+	quest.set(property, value)
+
+func get_quest_property(quest_id: int, property: String) -> Variant:
+	
+	# Find the quest
+	var quest: Quest = __get_quest_by_id(quest_id)
+
+	if quest == null: return null
+
+	# Now check if the quest has the property
+	var property_found : bool = __quest_has_property(quest, property)
+
+	# Return if the property was not found
+	if not property_found: return null
+
+	# Finally we get the value
+	return quest.get(property)
+
+func __quest_has_property(quest: Quest, property: String) -> bool:
+	# If the property is null -> we return
+	if property == null: return false
 
 	var was_property_found: bool = false
 	# Then we check if the property is present
 	for p in quest.get_property_list():
 		if p.name == property:
 			was_property_found = true
 			break
+			
+	return was_property_found
 
-	# Return if the property was not found
-	if not was_property_found: return
+func __get_quest_by_id(quest_id: int) -> Quest:
+	var quest: Quest = null
 
-	# Finally we set the value
-	quest.set(property, value)
+	# Find the quest
+	for pools in get_all_pools():
+		if pools.get_quest_from_id(quest_id) != null:
+			quest = pools.get_quest_from_id(quest_id)
+			
+	return quest
```

Comment:
I'd rename these methods removing one of the underscores to follow godot code style conventions.
Like this:
```gdscript
func _quest_has_property(...)
```